src/agentscope/memory/temporary_memory.py

```diff
@@ -46,14 +47,28 @@ def add(
         if memories is None:
             return
 
-        if not isinstance(memories, list):
+        if not isinstance(memories, Sequence):
             record_memories = [memories]
         else:
             record_memories = memories
 
         # if memory doesn't have id attribute, we skip the checking
         memories_idx = set(_.id for _ in self._content if hasattr(_, "id"))
         for memory_unit in record_memories:
+            if not issubclass(type(memory_unit), MessageBase):
+                try:
+                    if (
+                        "name" in memory_unit
+                        and memory_unit["name"] == "thought"
+                    ):
+                        memory_unit = Tht(**memory_unit)
+                    else:
```

Comment:
Add an `elif` entry for `PlaceholderMessage`

```
elif isinstance(memory_unit, PlaceholderMessage)
    memory_unit.update_value()
    memory_unit = Msg(**memory_unit)
```

Reply:
This `elif` won't be trigger because of the outter if-check.
Added another `if` for `PlaceholderMessage`.